### ISSUE TITLE
[C#] Implement new Concept Exercise: pointers

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/csharp#1447]
This issue describes how to implement the `pointers` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student advanced aspects of the Concept of Pointers in C#.

## Learning objectives

- Know what pointers are.
- Know how to define and use pointer types.
- Know how to use the `unsafe` keyword and pointer operators.
- Know how to use the `fixed` keyword.

## Out of scope

- Debugging memory leaks.

## Concepts

This Concepts Exercise's Concepts are:

- `pointers`: know what pointers are; know how to define and use pointer types; know how to use the `unsafe` keyword and pointer operators; know how to use the `fixed` keyword.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `memory-allocation`: know the basics of how memory works in C#, including reference and value type memory and garbage collector aspects.

### Hints

- [Pointer types][pointer-types]: introduction to pointer types.
- [Unsafe code and pointers][unsafe-code-and-pointers]: introduction to unsafe code and pointers.
- [`fixed` keyword][fixed]: how to work with the `fixed` keyword.
- [`unsafe` keyword][unsafe]: how to work with pointers in an `unsafe` context

### After

- [Pointer types][pointer-types]: introduction to pointer types.
- [Unsafe code and pointers][unsafe-code-and-pointers]: introduction to unsafe code and pointers.
- [`fixed` keyword][fixed]: how to work with the `fixed` keyword.
- [`unsafe` keyword][unsafe]: how to work with pointers in an `unsafe` context

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/concepts/pointer.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[pointer-types]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/unsafe-code-pointers/pointer-types
[unsafe-code-and-pointers]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/unsafe-code-pointers/
[fixed]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/unsafe-code-pointers/fixed-size-buffers
[unsafe]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/unsafe-code-pointers/pointer-types
